### PR TITLE
mpercolate api should serialise start time 

### PR DESCRIPTION
**Pull request by martijnvg**
PR for #15908 

(when back porting to other branches I'll add version checks for serialisation)

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java [hunk 1]
```diff
@@ -160,12 +160,8 @@ public void readFrom(StreamInput in) throws IOException {
             items = new ArrayList<>(size);
             for (int i = 0; i < size; i++) {
                 int slot = in.readVInt();
-                OriginalIndices originalIndices = OriginalIndices.readOriginalIndices(in);
-                PercolateShardRequest shardRequest = new PercolateShardRequest(new ShardId(index, shardId), originalIndices);
-                shardRequest.documentType(in.readString());
-                shardRequest.source(in.readBytesReference());
-                shardRequest.docSource(in.readBytesReference());
-                shardRequest.onlyCount(in.readBoolean());
+                PercolateShardRequest shardRequest = new PercolateShardRequest();
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java]: don't we need to read the original indices anymore here? not sure, but if that's the case we may be able to remove that specific PercolateShardRequest constructor.

Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java]: good point. We need to take the original indices into account. Let me change this.

Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java]: I think this change is ok and we do take the original indices into account. We do this when we create `PercolateShardRequest` instance in TransportMultiPercolateAction at line 206.

Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java]: you are right, the original indices are read/written in the super class, good! You can then remove the PercolateShardRequest that takes shardId and originalIndices as arguments I think.

Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java]: I tried to remove it but unit tests are using it (PercolateDocumentParserTests), the percolator document parser relies on the shard id to be set on the percolate shard request. So lets keep it?

Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java]: I think it's odd to have a public constructor for a test only... remove the OriginalIndices parameter at least? it's always set to null I think.

**Comment by martijnvg** [comment]
@javanna I've updated this PR.

**Comment by javanna** [comment]
LGTM

**Pull request merged**